projects/storefrontlib/src/recipes/storefront.module.ts

```diff
@@ -44,7 +43,6 @@ import { StorefrontFoundationModule } from './storefront-foundation.module';
     ProductDetailsPageModule,
     ProductListingPageModule,
     ExternalRoutesModule.forRoot(),
-    ConfiguratorCommonsModule.forRoot(),
```

Comment:
It's good we remove it, since the Configurator feature should be driven by the environment specific imports (`productconfig.feature.ts`), not the basic StorefrontModule. That said IMO we should discuss the main folders hierarchy:

IMO 
- `InteractiveConfigurationModule` should not import main `CommonConfiguratorComponentsModule`
- `OverviewModule` should not import main `CommonConfiguratorModule`
- `CommonConfiguratorModule` already imports `CommonConfiguratorComponentsModule` which is correct

In `productconfig.feature.ts` we should import those 3 modules:
- master `CommonConfiguratorModule` and
- `VariantConfiguratorModule`
- `TextfieldConfiguratorModule`

Reason:
It's OK we import in multiple places the same one module that declares and exports components/pipes/directives if only this place consumes those _declarables_. But I believe it's not the case for the `InteractiveConfigurationModule`, who imports `CommonConfiguratorComponentsModule`. (please note `CommonConfiguratorComponentsModule` is also imported in `CommonConfiguratorModule`)
Moreover I'd like to avoid importing in multiple places a module that provides something (i.e. reducers, other tokens). That would make customers' extensibility harder (please imagine a cusotmer who wants to get rid of some deeply-imported module by dismantling out modules and re-assembling; when a module with providers is imported in 2 places, customer has more work). `CommonConfiguratorModule` is such a module providing many stuff under the hood. That's why I'd like to import it once, shallow in the app.module (via `productconfig.feature.ts`).

---
Side note: I observed a lot of modules are private (even made theta-like-prefixed on purpose). IMO we should make them public anyway (for customers who want to dismanle and re-assemble modules in a custom way). If intermediate modules are private, customers cannot import them when re-assembling modules structure.

Reply:
Ok, got your point. One of my reasons was that if a project wanted to consume the variant configurator, they shouldn't be forced to care about the common configurator stuff, but the vc module should establish this relation itself.
But I see that this wouldn't work out even if we put the dependency in the variantConfiguratorModule (as e.g. if a customer would load the CPQ configurator, we would end up consuming the common configurator 2 times which from your answer I take is a thing we need to avoid).

Reply:
With regards to private exposure: Here I wasn't not sure, because I thought we should be 'conservative' with regards to API exposure. Does this exposure have anything to do with the classification of changes (breaking/non breaking)?
Will remove the private (theta) exposure

Reply:
> Does this exposure have anything to do with the classification of changes (breaking/non breaking)?

Yes. We should not do breaking changes in the module in a minor version. The [anti-breaking-change docs](https://sap.github.io/spartacus-docs/breaking-changes/#avoiding-breaking-changes) says:
_Removing anything from Angular modules, such as imports, providers, declarations, entryComponents, or exports._

Reason:
Exported module can be imported exclusively by a customer. A module comes with its behavior when being imported by a customer (i.e. the module provides some token for a customer OR it imports another module who provides something). In the end of the day, the customer imports a module to get some technical functionality. 

Most customers don't care, but those who use Spartacus in parts - they care (they dismantle our main modules and re-assemble only parts that they need).

If you think you'll likely re-arrange modules after your release, you may consider keeping them private.